metaflow/plugins/kubernetes/kubernetes.py

```diff
@@ -178,7 +179,7 @@ def create_job(
         job = (
             KubernetesClient()
             .job(
-                generate_name="t-",
+                generate_name="t-{uid}".format(uid=str(uuid.uuid4())),
```

Comment:
another approach here would be to rely on the existing id's for namespacing purposes (added benefit of making the pod name more descriptive). e.g.
```python
generate_name="t-{run_id}-{step_name}-{task_id}-".format(run_id=run_id, step_name=step_name, task_id=task_id),
```
This is similar to what is happening with Argo Workflows pod naming.

there's a max length of 63 characters for the name, and with a suffix of 5 characters there might be truncation in some cases. From what I've understood, the truncation only happens for the generatedName part, so we would still get a random suffix.

Reply:
alternatively we can just chop up the first one or two parts of the uuid, as we're still receiving the suffix from the apiserver on top of it.

Reply:
I like adding this to the name 👍

Potential issue:
1. If the step name is super long, then `truncate("t-{run_id}-{step_name}-{task_id}")` could end up not containing the task id, and thus we would have massive collisions on branching steps. So we would need something like `"t-{truncate(run_id)}-{truncate(step_name)}-{task_id}")` instead.
2. What if you have two or more metadata services running (I would even try to imagine why, but I think we have to assume people will do it), then the run_id, step_name, and task_id might be equal. I know it is probably unlikely, but I still kind of like having the 4 random characters to be on the safe side. 4 should be enough at this point :D

Unimportant side questions: What does the `t-` mean? And why not use f-strings?

Reply:
to introduce enough entropy - how about simply doing `generate_name="t-{entropy}".format(entropy=secrets.token_hex(2))`. This should be good enough for a vast majority of use cases till kubernetes address the [collision problem](https://github.com/kubernetes/kubernetes/issues/115489) on their end.

Reply:
The `token_hex` seems like a good way forward. Doesn't stretch the id's that much while providing enough entropy.

My earlier note about embedding more info to the pod id is a bit moot; as noted it is subject to truncation, and the relevant bits are recorded in pod annotations in full detail.

Reply:
The context was slightly different but we did face a similar issue here at Netflix and we solved it by doing something like this: 
- take 18 characters of a sha1 hash (or something like it)
- convert it to a smaller 12 characters using base64 encoding
- this gives you 72 bits of entropy and if you look here: https://kevingal.com/apps/collision.html, that is good enough for a billion tasks (well you have a 1e-6 chance of collision which is good enough for me :) ).

I don't fully understand how long lived the job name is but 16 bits of entropy (2 bytes for token_hex) seems low and you are likely to get a collision for 12000 jobs (one of the use cases brought on slack). The "trick" of using base64 encoding gives you higher entropy for fairly low cost (you just have to pick your other two characters carefully in case of naming conventions but that's not impossible). Not saying you need the full 72 bits but 16 feels low.

Reply:
the configurable part here is just a prefix supplied to the Kubernetes apiserver, it will still generate a 5-character suffix which it appends to it so overall it should be sufficient (`(16**4)*(40**5)` items if my midnight math is correct)

Reply:
Ah, I had missed the addition of 5 characters. If those characters have a 6 bit entropy (which covers alphanumerical strings using the full 52 characters and 10 digits) that would be 46 bits of entropy that would definitely be better :) (chances of collision under 1% for 1M jobs).

Reply:
The 5 characters provided by Kubernetes are all lowercase, thus 26 + 10 digits, I believe. So there are 36^5 total possible outcomes? With an additional 16 bits of entropy, I could run 100.000 jobs per day for 7 days and that would give me a 0.06% chance of collision - small enough. However, is 100.000 jobs per day an acceptable scaling limit? If I run 10x that I would get a collision probability of around 1%, which is too high in my point of view.

Personally, at my current company, I would be happy with 16 bits of entropy. But if we want to support more branching-heavy tasks I would vote for a few more bits of entropy.

```python
n_possible_job_names = (36**5) * (2**16) 
n_jobs = 7 * 100000
round(1 - np.prod([(n_possible_job_names - j) / n_possible_job_names for j in range(n_jobs)]), 2)
# = 0.06
```

Reply:
Perhaps a unix timestamp would be an option?

Reply:
Timestamps would be good if we could guarantee at most one job per second. But that is not really the case here.

As I see it, either we go with:

**1. Complete randomness**. In which case we should try to be on the safe side regarding entropy, eg.
```python
generate_name="t-{entropy}".format(entropy=secrets.token_hex(4))
```
which will give us around 0.01% chance of collisions with 7M jobs within 7 days TTL. 

**2. Some kind of increment like the task id**. Plus a little entropy to allow for eg. having two metadata services. So could be:
```python
generate_name="t-{task_id}-{entropy}".format(entropy=secrets.token_hex(2), task_id=task_id)
```

Reply:
Wouldn't the the random 5 character suffix (generated via`generateName`) mean you wouldn't need to to be restricted to one job per second?

Reply:
Sorry. You are right @roofurmston. Keep forgetting that. Then a UNIX timestamp should not be a problem.

Reply:
Since this is a temporary solution until the Kubernetes folks fix this, @savingoyal's simple solution seem sufficient